pkgconf>=2.4.0 errors out if the pc files of Requires.private are not found

--- "a/subprojects/meson-tests/common/44 pkgconfig-gen/meson.build"
+++ "b/subprojects/meson-tests/common/44 pkgconfig-gen/meson.build"
@@ -35,6 +35,10 @@
 
 # generate for systems without a glib pc file
 pkgg.generate(name: 'glib-2.0', dataonly: true)
+if v.version_compare('>=2.4.0')
+    pkgg.generate(name: 'gio-2.0', dataonly: true)
+    pkgg.generate(name: 'gobject-2.0', dataonly: true)
+endif
 
 pkgg.generate(
     libraries: [lib, '-lz'],
